nautobot/core/jobs/__init__.py

```diff
@@ -0,0 +1,115 @@
+import os
+
+from django.conf import settings
+
+from nautobot.core.celery import register_jobs
+from nautobot.extras.choices import LogLevelChoices
+from nautobot.extras.datasources import (
+    ensure_git_repository,
+    git_repository_dry_run,
+    refresh_datasource_content,
+)
+from nautobot.extras.jobs import Job, ObjectVar
+from nautobot.extras.models import GitRepository
+
+
+name = "System Jobs"
+
+
+class GitRepositoryPullAndRefreshData(Job):
+    """
+    System job to clone and/or pull a Git repository, then invoke refresh_datasource_content().
+    """
+
+    repository = ObjectVar(
+        description="Git Repository to pull and refresh",
+        label="Git Repository",
+        model=GitRepository,
+    )
+
+    class Meta:
+        name = "Git Repository: Sync"
+        has_sensitive_variables = False
+
+    def run(self, repository):
+        job_result = self.get_job_result()
+        user = job_result.user
+
+        job_result.log(
+            f'Creating/refreshing local copy of Git repository "{repository.name}"...',
+            logger=self.logger,
+        )
+
+        try:
+            if not os.path.exists(settings.GIT_ROOT):
+                os.makedirs(settings.GIT_ROOT)
```

Comment:
I wonder if we should consider removing this... `GIT_ROOT` should preexist IMHO.

Reply:
Yeah, I think it is in fact bootstrapped with `nautobot-server` runs.

Reply:
This is in `nautobot.core.cli._configure_settings()`:

```py
    if not os.path.exists(settings.GIT_ROOT):
        os.makedirs(settings.GIT_ROOT)
```